doc/plugins.dox

```diff
@@ -186,59 +195,143 @@ Let's start with the very basic headers and add more later if necessary.
+it important that you use this function to register your plugin, making it available to the user.
+Geany will pass a pointer to a GeanyPlugin instance which acts as a unique handle to your plugin.
+Use this pointer for registering and later API calls. Registering the plugin consists of a number
+of steps:
+
+  1. Filling GeanyPlugin::info with metadata
+    - @ref PluginInfo::name : The name of your plugin
+    - @ref PluginInfo::description : A brief description.
+    - @ref PluginInfo::version : The current version.
+    - @ref PluginInfo::author : Your contanct information.
+    .
+    Filling all of them is recommended to provide the best user experience, but only the name is truly
+    mandatory.
+  2. Filling GeanyPlugin::funcs with function pointers
+    - @ref GeanyPluginFuncs::init : an initialization function
+    - @ref GeanyPluginFuncs::cleanup : a finalization function
+    - @ref GeanyPluginFuncs::configure : a function that provides configuration (optional)
+    - @ref GeanyPluginFuncs::help : a function that provides help (optional)
+    - @ref GeanyPluginFuncs::callbacks : a pointer to an array of PluginCallback (optional).
+    .
+    @a init and @a cleanup are mandatory, the other ones depend on how advanced your plugin is.
+    Furthermore, the @a init and @a cleanup
+    pair is called when the user activates your plugin in the Plugin Manager. So use these to
+    do advanced initialization and teardown as to not waste resources when the plugin is not even
+    enabled.
+  3. Registering for real by calling @a geany_plugin_register(). You must pass the GeanyPlugin
+    pointer that your received and filled out as above. geany_plugin_register() also takes
+    various version numbers which are detailed below. Please <b>use the macro @a
+    GEANY_PLUGIN_REGISTER()</b> where you only specify the minimum API version. Please also
+    <b>observer the return value</b> of geany_plugin_register(). Geany may refuse to load your
+    plugin due to incompatibilities, you should then abort any extra setup.
+  4. Optionally setting user data that is passed to GeanyPlugin::funcs with geany_plugin_set_data().
+    Here you can set a data pointer that is passed back to your functions called by Geany. This
+    allows to avoid global variables which may be useful. It also allows to set instance pointer
+    to objects in case your plugin isn't written in pure C, enabling you to use member functions
+    as plugin functions.
+    You may also call this function later on, for example in your @ref GeanyPluginFuncs::init
+    routine to defer costly allocations to when the plugin is actually activated by the user.
+
+
+@subsection versions On API and ABI Versions
+As previously mentioned @a geany_plugin_register() takes a number of versions as arguments:
+  1. api_version
+  2. min_api_version
+  3. abi_version
+
+These refer to Geany's versioning scheme to manage plugin compatibility. It follows conrete rules:
```

Comment:
"conrete" s/b at least "concrete", but I don't think it's the right word.
Maybe "The API and ABI rules are as follows"?  not fancy but still better IMO